tests/unit/cmd/sca/test_scan.py

```diff
@@ -172,3 +173,67 @@ def test_sca_scan_all_cmd(
 CVE IDs: CVE-2023-30608"""
         in result.stdout
     )
+
+
+@my_vcr.use_cassette("test_sca_scan_pre_commit_no_arg.yaml")
+def test_sca_scan_pre_commit_no_arg(tmp_path, cli_fs_runner: CliRunner) -> None:
+    """
+    GIVEN a repository
+    WHEN running the sca scan pre-commit command with no argument
+    THEN the return code is 0
+    """
+    repo = Repository.create(tmp_path)
+    repo.create_commit()
+
+    with cd(str(tmp_path)):
+        result = cli_fs_runner.invoke(
+            cli,
+            ["sca", "scan", "pre-commit"],
+        )
+        assert result.exit_code == ExitCode.SUCCESS
+        assert "No SCA vulnerability added." in result.stdout
+
+
+@my_vcr.use_cassette("test_sca_scan_pre_commit_with_added_vulns.yaml")
+def test_sca_scan_pre_commit_with_added_vulns(
+    tmp_path, cli_fs_runner: CliRunner, pipfile_lock_with_vuln
+) -> None:
+    """
+    GIVEN a repository with a new file with vulns
+    WHEN running the sca scan pre-commit command
+    THEN we get the ExitCode.SCAN_FOUND_PROBLEMS
+    THEN the text output contains the added vulnerabilities
+    """
+
+    # Create a repo
+    repo = Repository.create(tmp_path)
+    repo.create_commit()  # Add an initial commit, if not, ref HEAD does not exist
+
+    # Writes a file with vulns
+    file_with_vulns = Path(tmp_path / "Pipfile.lock")
+    file_with_vulns.write_text(pipfile_lock_with_vuln)
+
+    # Add it to git
+    repo.add(str(file_with_vulns))
+
+    with cd(str(tmp_path)):
+        result = cli_fs_runner.invoke(
+            cli,
+            [
+                "sca",
+                "scan",
+                "pre-commit",
+                str(tmp_path),
+            ],
+        )
+
+        assert result.exit_code == ExitCode.SCAN_FOUND_PROBLEMS
+        assert "> Pipfile.lock: 1 incident detected" in result.stdout
+        assert (
+            """
+Severity: Medium
+Summary: sqlparse contains a regular expression that is vulnerable to Regular Expression Denial of Service
+A fix is available at version 0.4.4
+CVE IDs: CVE-2023-30608"""
+            in result.stdout
+        )
```

Comment:
would it be possible to add some other added/removed vulns? These asserts are the same as for scan all

Reply:
Added ! Good point, I realized that the text associated with removed vulns was "incident detected". Modified.